generators/typescript/utils/abstract-generator-cli/src/AbstractGeneratorCli.ts

```diff
@@ -123,7 +123,12 @@ export abstract class AbstractGeneratorCli<CustomConfig> {
                     });
                 },
                 downloadFiles: async () => {
-                    if (this.outputSourceFiles(customConfig)) {
+                    if (this.outputProjectFiles(customConfig)) {
+                        await typescriptProject.copyProjectAsZipTo({
+                            logger,
+                            destinationZip
+                        });
```

Comment:
This is the key to outputting the project files locally. Because this option outputs the entire project, the `outputSourceFiles` option is essentially implied since the `src` directory is included in the project. That's why this should take priority.